facad1ng.py

```diff
@@ -57,61 +59,86 @@ def print_banners():
+
+    if len(keywords) > max_length:
+        raise ValueError("Input string exceeds the maximum allowed length.")
+
     return "-".join(keywords.split())
 
 # Input from the user with validation
-while True:
-    web_url = input(f"{G}Enter the original link {W}(ex: https://www.ngrok.com): {W}")
-    try:
-        validate_web_url(web_url)
-        break
-    except ValueError as e:
-        print(e)
-
-while True:
-    custom_domain = input(f"\n{Y}Enter your custom domain {W}(ex: gmail.com): {W}")
-    try:
-        validate_custom_domain(custom_domain)
-        break
-    except ValueError as e:
-        print(e)
-
-while True:
-    phish = input(f"\n{C}Enter phishing keywords {W}(ex: free-stuff, login): {W}")
-    phish = format_phish_keywords(phish)
-    if " " not in phish:
-        break
-    else:
-        print("Phishing keywords should not contain spaces. Use '-' to separate them.")
-
-# Prepare the data for the request
-data = {
-    'url': web_url,
-    'shorturl': '',
-}
-
-# Shorten the original URL with multiple URL shorteners
-short_urls = [shortener.short(web_url) for shortener in shorteners]
-
-# Mask the URLs with custom domain and phishing keywords
-def mask_url(domain, keyword, url):
-    # Use urlparse to properly split the URL
-    parsed_url = urlparse(url)
-
-    # Reconstruct the URL with the custom domain and phishing keyword
-    return f"{parsed_url.scheme}://{domain}-{keyword}@{parsed_url.netloc}{parsed_url.path}"
-
-# Print the results
-print(f"\n{Y}Original URL:{W}", web_url, "\n")
-print(f"{G}[~] {R}Masked URL (using multiple shorteners):{W}")
-for i, short_url in enumerate(short_urls):
-    masked_url = mask_url(custom_domain, phish, short_url)
-    print(f"{G}╰➤ {C}Shortener {W} {i + 1}: {masked_url}")
+try:
+    while True:
+        web_url = input(f"{G}Enter the original link {W}(ex: https://www.ngrok.com): {W}")
+        try:
+            validate_web_url(web_url)
+            break
+        except ValueError as e:
+            print(e)
+
+    while True:
+        custom_domain = input(f"\n{Y}Enter your custom domain {W}(ex: gmail.com): {W}")
+        try:
+            validate_custom_domain(custom_domain)
+            break
+        except ValueError as e:
+            print(e)
+
+    while True:
+        phish = input(f"\n{C}Enter phishing keywords {W}(ex: free-stuff, login): {W}")
+        phish = format_phish_keywords(phish)
+        if " " not in phish:
+            break
+        else:
+            print("Phishing keywords should not contain spaces. Use '-' to separate them.")
+
+    # Prepare the data for the request
+    data = {
+        'url': web_url,
+        'shorturl': '',
+    }
+
+    # Shorten the original URL with multiple URL shorteners
+    short_urls = [shortener.short(web_url) for shortener in shorteners]
+
+    # Mask the URLs with custom domain and phishing keywords
+    def mask_url(domain, keyword, url):
+
+        encoded_url = base64.b64encode(url.encode()).decode()
```

Comment:
don't perform base64 encoding

Reply:
This was your idea to hide the vendor name. if not base64 how you gonna hide the shortened URL

Reply:
the problem with base64 is that the final URL you get won't redirect you to the phishing page. 

For example: this original link `https://ngrok.com`, when converted to base64 looks like this `https://gmail.com-bG9naW4=@aHR0cHM6Ly90aW55dXJsLmNvbS9wcWpzcmU1`. Now the single main issue with this type of url is, when the user pastes it into the browser, the request won't go to the ngrok.com. The base64 encoding of the given url didn't fit into the picture as we imagined it, she checked with the URL encoding as well where special chars gets encoded but it didn't work either. 

We are still working on this, will let you know the possible solutions.

Reply:
we can use server-side redirection script to achieve our goal but for this we have to run web server like flask

```
def encode_and_redirect(original_url):
    """
    Encode the original URL and redirect to the server-side endpoint
    """
    s = pyshorteners.Shortener()
    shortened_url = s.tinyurl.short(original_url)
    encoded_url = urlparse(shortened_url).path[1:]
    return redirect(f'/redirect/{encoded_url}')


@app.route('/redirect/<path:encoded_url>')
def redirect_to_original(encoded_url):
    """
    Redirect to the original URL using the decoded URL
    """
    decoded_url = encoded_url.encode('utf-8')
    return redirect(decoded_url)


if __name__ == '__main__':
    print_banners()
    app.run(debug=True)
```

In this example, a Flask web server is used to handle redirection. When you encode a URL using this script, it will create a shortened URL like http://yourserver.com/redirect/encoded_url, and when users visit this shortened URL, they will be redirected to the original URL. This way, you can hide the original URL while still allowing proper redirection.

Not a fan of this method but this can be a solution.